### PR TITLE
fix lastname processing return; disable debug printing: add some common abbreviation for givennames.

### PR DESCRIPTION
**Pull request by random-ao**
- Fix lastname processing return to fix truncation of data.
- Disable debug printing, until we have argv support, to make performance great again.

stats:
records: input 1451073 -> output 163419 = 11%
family names: total 496063; known: 237771 = 48%
firstnames: total 591416; known: 289955 = 49%

records: input 1451073 -> output 304299 = 21%
family names: total 496063; known: 237771 = 48%
firstnames: total 799970; known: 548084 = 69%

### CODE REVIEW

File: src/process.py [hunk 1]
```diff
@@ -30,6 +30,9 @@ def __init__(self, cachedir):
         self.good_firstname_count = 0
         self.bad_firstname_count = 0
         self.unknown_firstnames = Counter()
+        self.firstnames_expansion_map = {
+          "alb.": "Albert"
```
Comment (pdaengeli) [inline src/process.py:34]: Isn't firstname abbreviations expansion a can of worms? At least Albrecht, Albertine, Alban, Alberte and Albin might think so ;)

I'd rather not introduce false precision here and would prefer to offer possible variants in the frontend only, e.g. in a tooltip when hovering over the abbreviated name (ideally querying a database of (abbreviated) German first names of the first half of the 20th century, but such a thing might not exist).

Comment (random-ao) [reply src/process.py:34]: duh, you're absolutely correct, I should have added Alb. to givennames.csv, as an accepted abbreviation. Will amend and ping this pr when done.

Comment (random-ao) [reply src/process.py]: updated

**Comment by brawer** [comment]
It might be good to keep the abbreviations in a separate file, with their potential expansions. For example, when eventually implementing search over this data, some mapping of abbreviations to expanded string would be super useful. Let me give it a try in a follow-up change.

**Comment by random-ao** [comment]
Thanks! I added them there since givennames.csv is already polluted - but I
too thought this is odd. (Another observation: dupes).

Splitting them out will also make it more straightforward to resolve clear
1:1 match abbrvs).

Sascha Brawer - notifications at github.com <
***@***.***> schrieb am
Mo., 15. Mai 2023, 14:40:

**Pull request merged**